test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -1227,6 +1219,29 @@ public static enum StandardAssert {
                 MacHelper.verifyUnsignedBundleSignature(cmd);
             }
         }),
+        MAC_RUNTIME_PLIST_JDK_KEY(cmd -> {
+            if (TKit.isOSX()) {
+                var appLayout = cmd.appLayout();
+                // Runtime directory may not exist.
+                // For example in AppImagePackageTest.testEmpty() case.
+                if (!Files.isDirectory(appLayout.runtimeDirectory())) {
+                    return;
+                }
```

Comment:
A missing runtime directory is not a regular situation.

In such a case, (I believe `AppImagePackageTest.testEmpty()` is the only one) test should explicitly exclude this assert. See how `AppImagePackageTest.testEmpty()` excludes the `RUNTIME_DIRECTORY` and other asserts. 

I suggest removing the test for the runtime directory from the assert body and adding `MAC_RUNTIME_PLIST_JDK_KEY` assert to the list of excluded asserts in the `AppImagePackageTest.testEmpty()` test.